test/govuk_component/taxonomy_sidebar_test.rb

```diff
@@ -13,53 +13,101 @@ def component_name
 
   test "renders a related items block" do
     render_component(
-      sections: [
+      items: [
         {
-          title: "Section title",
-          items: [
+          title: "Item title",
+          url: "/item",
+          description: "An item",
+        },
+        {
+          title: "Other item title",
+          url: "/other-item",
+          description: "Another item",
+        },
+      ]
+    )
+
+    assert_select ".govuk-taxonomy-sidebar > :nth-of-type(1) h2", text: "More about Item title"
+    assert_link_with_text_in(".govuk-taxonomy-sidebar > :nth-of-type(1)", "/item", "See everything in Item title")
+    assert_select(".govuk-taxonomy-sidebar > :nth-of-type(1) p", "An item")
+
+    assert_select ".govuk-taxonomy-sidebar > :nth-of-type(2) h2", text: "More about Other item title"
+    assert_link_with_text_in(".govuk-taxonomy-sidebar > :nth-of-type(2)", "/other-item", "See everything in Other item title")
+    assert_select(".govuk-taxonomy-sidebar > :nth-of-type(2) p", "Another item")
+  end
+
+  test "renders related content for taxons" do
+    render_component(
+      items: [
+        {
+          title: "Item title",
+          url: "/item",
+          description: "An item",
+          related_content: [
+            {
+              title: "Related link 1",
+              link: "/related-link-1",
+            },
             {
-              title: "Item title",
-              url: "/item",
-              description: "An item",
+              title: "Related link 2",
```

Comment:
Can we remove data for Related link 2 and 3 if we aren't making assertions about them below? Unnecessary test setup.